types/plugin.d.ts

```diff
@@ -125,8 +125,11 @@ declare namespace fastifyCookie {
     signed?: boolean;
   }
 
+  type HookType = 'onRequest' | 'preParsing' | 'preValidation' | 'preHandler'  | 'preSerialization';
+
   export interface FastifyCookieOptions {
     secret?: string | string[] | Signer;
+    hook?: HookType | false;
```

Comment:
Would you mind adding a test for the new type?

Reply:
I can, but what is there to check?

Reply:
Basically he want that you add following lines into plugin.test-d.ts

```typescript
const appWithHook: FastifyInstance = fastify();

appWithHook.register(cookie, { hook: false });
appWithHook.register(cookie, { hook: 'onRequest' });
appWithHook.register(cookie, { hook: 'preHandler' });
appWithHook.register(cookie, { hook: 'preParsing' });
appWithHook.register(cookie, { hook: 'preSerialization' });
appWithHook.register(cookie, { hook: 'preValidation' });
expectError(appWithHook.register(cookie, { hook: true }))
expectError(appWithHook.register(cookie, { hook: 'false' }))
```

Reply:
done